lib/awful/util.lua

```diff
@@ -40,8 +40,17 @@ local displayed_deprecations = {}
 -- @param[opt] see The message to a new method / function to use.
 -- @tparam table args Extra arguments
 -- @tparam boolean args.raw Print the message as-is without the automatic context
+-- @tparam integer args.deprecated_in Print the message only when Awesome's major version is equal to or greater than deprecated_in
 function util.deprecate(see, args)
     args = args or {}
+    -- If args.deprecated_in is defined, only print deprecation warning after the given major version
+    if args and args.deprecated_in then
+        local cur_ver, dep_ver = awesome.version, "v" .. tostring(args.deprecated_in) .. ".0"
```

Comment:
```diff
diff --git a/lib/awful/util.lua b/lib/awful/util.lua
index 2bec8ba..ee7e10f 100644
--- a/lib/awful/util.lua
+++ b/lib/awful/util.lua
@@ -44,10 +44,9 @@ local displayed_deprecations = {}
 function util.deprecate(see, args)
     args = args or {}
     -- If args.deprecated_in is defined, only print deprecation warning after the given major version
-    if args and args.deprecated_in then
-        local cur_ver, dep_ver = awesome.version, "v" .. tostring(args.deprecated_in) .. ".0"
-        -- Alternatively think as: if awesome.version >= deprecated_version then print the deprecated warning
-        if cur_ver < dep_ver then
+    if args.deprecated_in then
+        local dep_ver = "v" .. tostring(args.deprecated_in) .. ".0"
+        if awesome.version < dep_ver then
             return
         end
     end
```

Here's the current change. So do we want to make `dep_ver = "v" .. tostring(args.deprecated_in)`? How do you want to format `dep_ver` to be less magic?